src/OptionsPresenter.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace OdbavTo\PresenterRoute;
+
+use Nette\Application\IResponse;
+use Nette\Application\Request;
+use Nette\Application\Responses\JsonResponse;
+use Nette\Http\Response as HttpResponse;
+
+class OptionsPresenter
+{
+	/**
+	 * @var \Nette\Http\Response
+	 */
+	private $httpResponse;
+
+	public function __construct(HttpResponse $httpResponse)
+	{
+		$this->httpResponse = $httpResponse;
+	}
+
+
+	public function __invoke(Request $request): IResponse
+	{
+		$methodsAllowed = $request->getParameter(RestRouteList::ALLOWED_METHODS_KEY);
+
+		if (empty($methodsAllowed)) {
+			$this->httpResponse->setCode(404);
```

Comment:
Please test it with `throw new BadRequestException()`, i think it is more describing than sending 404 json response - exceptions are handled by error presenter and it will set 404 by itself and allow custom response depending on project.